network/h2c.go

```diff
@@ -27,6 +27,18 @@ import (
 	"golang.org/x/net/http2/h2c"
 )
 
+// NewServerTLS is same as NewServer but allows configuring the server's TLSConfig
+func NewServerTLS(addr string, h http.Handler, tlsConfig *tls.Config) *http.Server {
+	//nolint:gosec
+	h1s := &http.Server{
+		Addr:      addr,
+		Handler:   h2c.NewHandler(h, &http2.Server{}),
+		TLSConfig: tlsConfig,
+	}
+
+	return h1s
+}
+
```

Comment:
Consider using the functional options pattern:

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

This would look like:

```golang
type ServerOption func(*http.Server)

func WithTLSConfig(tlsConfig *tls.Config) ServerOption {
  return func(s *http.Server) {
    s.TLSConfig = tlsConfig
  }
}

func NewServer(addr string, h http.Handler, opts ...ServerOption) *http.Server {
  h1s := ...

  for _, opt := range opts {
    opt(h1s)
  }

  return h1s
}
```

Among other things, it will ensure that the `NewServerTLS` and other implementations also implement the slowloris attack mitigation on line 47.

Also, it will avoid creating many `NewServerFoo` implementations, at the host of writing:

```
srv = NewServer(":8080", myHandler, WithTLSConfig(myTLS))
```